src/hotspot/share/gc/shared/cardTable.cpp

```diff
@@ -78,9 +77,8 @@ CardTable::CardTable(MemRegion whole_heap) :
 void CardTable::initialize(void* region0_start, void* region1_start) {
   size_t num_cards = cards_required(_whole_heap.word_size());
 
-  // each card takes 1 byte; + 1 for the guard card
-  size_t num_bytes = num_cards + 1;
```

Comment:
I think it's better not to remove the `num_bytes` var. Otherwise, one have keep `one card == one byte` in mind everywhere #bytes is expected but #cards is used.

Reply:
Do you want to leave `size_t num_bytes = num_cards;`? May it be strange? We already have the comment to clarify it.

Reply:
> Do you want to leave size_t num_bytes = num_cards;?

Yes. That comment is for why the conversion factor is one and the unit conversion is explicit. Otherwise, the conversion occurs as the arg is passed `compute_byte_map_size`, which can be confusing and error-prone, IMO.

Reply:
Ohh, the method `compute_byte_map_size` takes the number of bytes as its parameter. Fixed.